examples/server/server.cpp

```diff
@@ -2842,7 +2888,10 @@ int main(int argc, char **argv)
     }
 
     if (sparams.metrics_endpoint) {
-        svr.Get("/metrics", [&](const httplib::Request&, httplib::Response& res) {
+        svr.Get("/metrics", [&](const httplib::Request& req, httplib::Response& res) {
```

Comment:
/metrics must not be protected. It does not contain data and it targets prometheus which does not support authentication.

Reply:
This is the purpose of the `key` query param.

Reply:
Could you link to a security protocol this query param implements ?

Reply:
Not sure what you mean. This is just passing in the same api/admin key in through `?key=` instead of the authorization header for ease in configuration. It's the same as the authorization header otherwise.

Reply:
Kindly, I meant where have you seen we can pass secret as query param in a protocol ? I am wondering if it is a security issue.
For example, in oauth2 implicit grant flow, this has been deprecated.

Reply:
Any form of authentication via a URL Query Param is considered bad practice and is inherently unsafe, even when using TLS/SSL. There's always a "gotcha" somewhere that exposes the authentication method. This is why it's usually done with a bearer token or jwt which is passed via an encrypted tunnel. You can find this all on OWASP and within the RFC Specs. This has to do with underlying mechanics of how GET and POST requests work. For a real authy method, you would use POST and not expose the Auth tokens via Query Params.

Reply:
The question was the opposite. We all know that this is not security here. That's why this PR is open... There is no need for beginners' explanations